tutor/env.py

```diff
@@ -342,6 +342,8 @@ def save(root: str, config: Config) -> None:
         save_all_from(src, os.path.join(root_env, dst), config)
 
     upgrade_obsolete(root)
+    # print("asdasd")
+    hooks.Actions.ENV_SAVED.do(root_env, config)
```

Comment:
The docs you add mention that modifying this `config` variable won't result in the configuration being resaved and that it should be treated as read-only.

However, the config object being passed into this function is handed to this function. I'm not sure what the lifetime is of this variable outside of the function, but presumably, the configuration could be modified by this function and affect later code execution, even if the result isn't saved.

Should we add in a bit of enforcement here, by cloning the object before passing it? If so, might we do an equality check between the config before and after the change and add a warning saying the changes will not be saved?